app/controllers/files/files_controller.rb

```diff
@@ -37,7 +38,12 @@ def destroy
     # make sure file belongs to current identity and delete file
     if @file.identity == @identity && @identity.user == current_user && @file.destroy
       @message = "File deleted."
-      @file = nil
+
+      if @file.identity.service == 'local'
+        File.delete(Rails.root + 'uploads/nil_class/heej/' + @file.name)
```

Comment:
Since the upload path is stored as the foreign ref it can be used. Something like this:
```ruby
File.delete(Rails.root + @file.foreign_ref)
```